File: 0700/main.ts
function searchBST(root: TreeNode | null, val: number): TreeNode | null {
    if (root == null){
        return null
    }
    if (root.val == val){
        return root
    }
    if (root.val < val){
        return searchBST(root.right, val)
    }
    return searchBST(root.left, val)
};

File: 0700/main.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
        
class Solution:
    def searchBST(self, root: TreeNode, val: int) -> TreeNode:
        if root == None:
            return None
        if root.val == val:
            return root
        if root.val < val:
            return self.searchBST(root.right, val)
        return self.searchBST(root.left, val)